ideascube/configuration/models.py

```diff
@@ -0,0 +1,76 @@
+from django.conf import settings
+from django.db import models
+
+from ideascube.models import JSONField
+
+from .errors import (
+    InvalidConfigurationValue,
+    InvalidStoredConfigurationValue,
+    NoSuchConfigurationKey,
+    NoSuchConfigurationNamespace,
+    )
+from .registry import REGISTRY
+
+
+class Configuration(models.Model):
+    class Meta:
+        ordering = ['-date']
+
+    namespace = models.CharField(max_length=40)
+    key = models.CharField(max_length=40)
+    value = JSONField()
+    actor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    date = models.DateTimeField(auto_now=True)
+
+    @staticmethod
+    def _get_key_registry(namespace, key):
+        try:
+            namespace_registry = REGISTRY[namespace]
+
+        except KeyError:
+            raise NoSuchConfigurationNamespace(namespace)
+
+        try:
+            return namespace_registry[key]
+
+        except KeyError:
+            raise NoSuchConfigurationKey(namespace, key)
+
+    @classmethod
+    def get(cls, namespace, key):
+        key_registry = cls._get_key_registry(namespace, key)
+        expected_type = key_registry['python_type']
+
+        config = cls.objects.get(namespace=namespace, key=key)
+
+        if not isinstance(config.value, expected_type):
+            raise InvalidStoredConfigurationValue(
+                config, expected_type, type(config.value))
+
+        return config
+
+    @classmethod
+    def get_value(cls, namespace, key):
+        try:
+            return cls.get(namespace, key).value
+
+        except cls.DoesNotExist:
+            return REGISTRY[namespace][key]['default']
```

Comment:
The thinking is that `Configuration.get` being a class method it returns a `Configuration` instance, whereas `get_value` is a useful shortcut to get only the value (which is an attribute of the instance) of that configuration.

The names seem expressive to me, but since I wrote the code I might not be the right judge.

However, both are useful in different contexts, so neither should be private.

I agree this is lacking documentation, but I've been kind of cooled down on writing docs since that time @yohanboniface asked me to remove some doc I had written, so now I wait for someone to ask for doc before writing it. :stuck_out_tongue_winking_eye: I'll add some here.